server/container/guice/guice-common/src/main/java/org/apache/james/utils/ConfigurationPerformer.java

```diff
@@ -19,9 +19,12 @@
 
 package org.apache.james.utils;
 
+import org.apache.james.lifecycle.api.Configurable;
 
-public interface ConfigurationPerformer {
+public interface ConfigurationPerformer<CONF extends Configurable> {
 
     void initModule() throws Exception;
 
+    Class<CONF> forClass();
```

Comment:
even better : you can now return a Set of Configurable and avoid splitting some ConfigurationPerformer.